plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestCdfWithManySplits.java

```diff
@@ -0,0 +1,141 @@
+
+    private static final String bucketName = "test-bucket-" + randomNameSuffix();
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Minio minio = closeAfterClass(Minio.builder().build());
+        minio.start();
+        minio.createBucket(bucketName);
+
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(testSessionBuilder()
+                        .setCatalog(DELTA_CATALOG)
+                        .setSchema(SCHEMA)
+                        .build())
+                .setExtraProperties(ImmutableMap.of(
+                        "query.schedule-split-batch-size", "1",
+                        "node-scheduler.max-splits-per-node", "1",
+                        "node-scheduler.min-pending-splits-per-task", "1"))
+                .build();
+        try {
+            queryRunner.installPlugin(new TpchPlugin());
+            queryRunner.createCatalog("tpch", "tpch");
+
+            queryRunner.installPlugin(new DeltaLakePlugin());
+            queryRunner.createCatalog(DELTA_CATALOG, DeltaLakeConnectorFactory.CONNECTOR_NAME, ImmutableMap.<String, String>builder()
+                    .put("hive.metastore", "file")
+                    .put("hive.metastore.catalog.dir", queryRunner.getCoordinator().getBaseDataDir().resolve("file-metastore").toString())
+                    .put("hive.metastore.disable-location-checks", "true")
+                    .put("hive.s3.aws-access-key", MINIO_ACCESS_KEY)
+                    .put("hive.s3.aws-secret-key", MINIO_SECRET_KEY)
+                    .put("hive.s3.endpoint", minio.getMinioAddress())
+                    .put("hive.s3.path-style-access", "true")
+                    .put("delta.enable-non-concurrent-writes", "true")
+                    .put("delta.register-table-procedure.enabled", "true")
+                    .buildOrThrow());
+
+            queryRunner.execute("CREATE SCHEMA " + SCHEMA + " WITH (location = 's3://" + bucketName + "/" + SCHEMA + "')");
+        }
+        catch (Throwable e) {
+            closeAllSuppress(e, queryRunner);
+            throw e;
+        }
+
+        return queryRunner;
+    }
+
+    @Test
+    public void testReadCdfChangesWithMaxSplitsSetToOne()
+    {
+        String tableName = "test_basic_operations_on_table_with_cdf_enabled_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " (page_url VARCHAR, domain VARCHAR, views INTEGER) WITH (change_data_feed_enabled = true)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES('url1', 'domain1', 1), ('url2', 'domain2', 2), ('url3', 'domain3', 3)", 3);
+        assertUpdate("INSERT INTO " + tableName + " VALUES('url4', 'domain4', 4), ('url5', 'domain5', 2), ('url6', 'domain6', 6)", 3);
+
+        assertUpdate("UPDATE " + tableName + " SET page_url = 'url22' WHERE views = 2", 2);
+        assertTableChangesQuery("SELECT * FROM TABLE(system.table_changes('test_schema', '" + tableName + "'))",
```

Comment:
```
trino:tiny> select * from table(system.table_changes('tiny', 'test1'));
 page_url | domain  | views |   _change_type   | _commit_version |      _commit_timestamp      
----------+---------+-------+------------------+-----------------+-----------------------------
 url1     | domain1 |     1 | insert           |               1 | 2023-07-21 08:35:51.846 UTC 
 url2     | domain2 |     2 | insert           |               1 | 2023-07-21 08:35:51.846 UTC 
 url3     | domain3 |     3 | insert           |               1 | 2023-07-21 08:35:51.846 UTC 
 url4     | domain4 |     4 | insert           |               2 | 2023-07-21 08:36:05.199 UTC 
 url5     | domain5 |     2 | insert           |               2 | 2023-07-21 08:36:05.199 UTC 
 url6     | domain6 |     6 | insert           |               2 | 2023-07-21 08:36:05.199 UTC 
 url22    | domain5 |     2 | update_postimage |               3 | 2023-07-21 08:36:20.594 UTC 
 url22    | domain2 |     2 | update_postimage |               3 | 2023-07-21 08:36:20.594 UTC 
 url2     | domain2 |     2 | update_preimage  |               3 | 2023-07-21 08:36:20.594 UTC 
 url5     | domain5 |     2 | update_preimage  |               3 | 2023-07-21 08:36:20.594 UTC 
(10 rows)

Query 20230721_084139_00016_jkxp6, FINISHED, 1 node
http://localhost:8080/ui/query.html?20230721_084139_00016_jkxp6
Splits: 3 total, 3 done (100.00%)
CPU Time: 0.0s total,     0 rows/s,     0B/s, 72% active
Per Node: 0.0 parallelism,     0 rows/s,     0B/s
Parallelism: 0.0
Peak Memory: 1.21KB
1.10 [0 rows, 0B] [0 rows/s, 0B/s]

trino:tiny> explain analyze select * from table(system.table_changes('tiny', 'test1'));
                                                                              Query Plan                                                                               
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Trino version: dev                                                                                                                                                    
 Queued: 255.83us, Analysis: 50.45ms, Planning: 9.71ms, Execution: 695.27ms                                                                                            
 Fragment 1 [SOURCE]                                                                                                                                                   
     CPU: 26.54ms, Scheduled: 39.07ms, Blocked 0.00ns (Input: 0.00ns, Output: 0.00ns), Input: 0 rows (0B); per task: avg.: 0.00 std.dev.: 0.00, Output: 10 rows (526B) 
     Output layout: [page_url, domain, views, _change_type, _commit_version, _commit_timestamp]                                                                        
     Output partitioning: SINGLE []                                                                                                                                    
     TableFunctionProcessor[name = table_changes, properOutputs = [page_url, domain, views, _change_type, _commit_version, _commit_timestamp]]                         
         Layout: [page_url:varchar, domain:varchar, views:integer, _change_type:varchar, _commit_version:bigint, _commit_timestamp:timestamp(3) with time zone]        
         CPU: 27.00ms (100.00%), Scheduled: 38.00ms (100.00%), Blocked: 0.00ns (?%), Output: 10 rows (526B)                                                            
         Input avg.: 0.00 rows, Input std.dev.: ?%                                                                                                                     
                                                                                                                                                                       
                                                                                                                                                                       
(1 row)
```
I find it a bit strange that the `TableFunctionProcessor` doesn't return the right amount of rows in `EXPLAIN ANALYZE`. Ideas?